Source/WebCore/rendering/svg/SVGTextMetricsBuilder.cpp

```diff
@@ -76,8 +76,18 @@ void SVGTextMetricsBuilder::advanceSimpleText()
 void SVGTextMetricsBuilder::advanceComplexText()
 {
     unsigned metricsLength = currentCharacterStartsSurrogatePair() ? 2 : 1;
-    m_currentMetrics = SVGTextMetrics::measureCharacterRange(*m_text, m_textPosition, metricsLength);
-    m_complexStartToCurrentMetrics = SVGTextMetrics::measureCharacterRange(*m_text, 0, m_textPosition + metricsLength);
+    float beforeWidth = 0;
+    float afterWidth = 0;
+
+    m_complexTextController->advance(m_textPosition, nullptr);
```

Comment:
Are you sure you don't need to check if `m_textPosition` is behind the current position of the complex text controller? The previous code was stateless, and you're introducing state into it now.